tests/testsuite/check.rs

```diff
@@ -1500,25 +1624,33 @@ fn check_unused_manifest_keys() {
         .build();
 
     p.cargo("check")
-        .with_stderr(
-            "\
+        .with_stderr_data(
+            format!(
+                "\
 [WARNING] unused manifest key: dependencies.dep.wxz
 [WARNING] unused manifest key: dependencies.foo.abc
 [WARNING] unused manifest key: dev-dependencies.foo.wxz
 [WARNING] unused manifest key: build-dependencies.foo.wxz
 [WARNING] unused manifest key: target.bar.build-dependencies.foo.wxz
 [WARNING] unused manifest key: target.cfg(windows).dependencies.foo.wxz
-[WARNING] unused manifest key: target.x86_64-pc-windows-gnu.dev-dependencies.foo.wxz
-[UPDATING] `[..]` index
+[WARNING] unused manifest key: target.{}.dev-dependencies.foo.wxz
+[UPDATING] `dummy-registry` index
 [LOCKING] 3 packages to latest compatible versions
 [DOWNLOADING] crates ...
-[DOWNLOADED] foo v0.1.0 ([..])
-[DOWNLOADED] dep v0.1.0 ([..])
-[CHECKING] [..]
-[CHECKING] [..]
-[CHECKING] bar v0.2.0 ([CWD])
-[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [..]
+[DOWNLOADED] foo v0.1.0 (registry `dummy-registry`)
+[DOWNLOADED] dep v0.1.0 (registry `dummy-registry`)
+[CHECKING] foo v0.1.0
+[CHECKING] dep v0.1.0
+[CHECKING] bar v0.2.0 ([ROOT]/foo)
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
 ",
+                if cfg!(all(target_os = "windows", target_env = "gnu")) {
+                    "[HOST_TARGET]"
+                } else {
+                    "x86_64-pc-windows-gnu"
+                }
+            )
+            .unordered(),
```

Comment:
`x86_64-pc-windows-gnu` is being modified to `[HOST_TARGET]` on windows with gnu. 😅 

FYI: https://github.com/rust-lang/cargo/actions/runs/9786239334/job/27020737537#step:11:4049
```
 --- Expected
++++ actual:   stderr
   1    1 | [WARNING] unused manifest key: dependencies.dep.wxz
   2    2 | [WARNING] unused manifest key: dependencies.foo.abc
   3    3 | [WARNING] unused manifest key: dev-dependencies.foo.wxz
   4    4 | [WARNING] unused manifest key: build-dependencies.foo.wxz
   5    5 | [WARNING] unused manifest key: target.bar.build-dependencies.foo.wxz
   6    6 | [WARNING] unused manifest key: target.cfg(windows).dependencies.foo.wxz
   7      - [WARNING] unused manifest key: target.x86_64-pc-windows-gnu.dev-dependencies.foo.wxz
   8    7 | [UPDATING] `dummy-registry` index
   9    8 | [LOCKING] 3 packages to latest compatible versions
  10    9 | [DOWNLOADING] crates ...
  11   10 | [DOWNLOADED] foo v0.1.0 (registry `dummy-registry`)
  12   11 | [DOWNLOADED] dep v0.1.0 (registry `dummy-registry`)
  13   12 | [CHECKING] foo v0.1.0
  14   13 | [CHECKING] dep v0.1.0
  15   14 | [CHECKING] bar v0.2.0 ([ROOT]/foo)
  16   15 | [FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
       16 + [WARNING] unused manifest key: target.[HOST_TARGET].dev-dependencies.foo.wxz
```

Reply:
Maybe we can just change to something like wasm32-wasip1 that would never conflict?

Reply:
Great idea!